sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -143,6 +144,22 @@ protected internal virtual object InstrumentOperation(Type operationType, object
             return operation;
         }
 
+        protected internal T InstrumentOperation<T>(T operation) where T : Operation =>
```

Comment:
I believe these methods were left off of here because we shouldn't be adding the OperationInstrumentor when we are not in playback mode and playback mode was assumed to be only for recorded tests.  We might want to come up a MockTestBase that allows us to maintain the enforcement of only applying the operation instrumentation during playback but at the same time allows us to override that only in the Mock scenario.  It would probably also be worth it to add a mock instrumentor to enforce that if your test base doesn't inherit from the mocktestbase then your transport shouldn't be a mocktransport and vice versa.

Reply:
These methods aren't called by default so I see no harm is refactoring them from `RecordedTestBase` to `ClientTestBase`. Could you clarify why you are concerned where they are defined when they have to be manually called?

Reply:
See https://github.com/Azure/azure-sdk-for-net/pull/27399#discussion_r821914946